addon/components/frost-object-details.js

```diff
@@ -1,5 +1,77 @@
 import Ember from 'ember'
+import layout from '../templates/components/frost-object-details'
+import _ from 'lodash'
 
 export default Ember.Component.extend({
-  classNames: ['frost-object-details']
+
+  _routing: Ember.inject.service('-routing'),
+
+  layout: layout,
+
+  classNames: ['frost-object-details'],
+
+  parentRouteName: Ember.computed('_routing.currentRouteName', function () {
+    let currentRouteName = this.get('_routing.currentRouteName')
+    if (_.includes(currentRouteName, '.views.')) {
+      return currentRouteName.substring(0, currentRouteName.indexOf('.views.'))
+    } else if (_.includes(currentRouteName, '.related.')) {
+      return currentRouteName.substring(0, currentRouteName.indexOf('.related.'))
+    }
+  }),
+
+  viewRoutes: Ember.computed('_routing', 'parentRouteName', function () {
+    let availableRoutes = Object.keys(this.get('_routing.router').router.recognizer.names)
```

Comment:
Eh, minimal savings, ignore me - you could make it a function call though, passing either '.views.' or '.related.' as the filter parameter